### PR TITLE
Adds Prod Stage

### PR DESCRIPTION
**Pull request by williamayerst**
#### 📲 What

Adds Prod Stage

#### 🤔 Why
		
Strategic requirement
		
#### 🛠 How
		
Standard GH pipeline configuration

#### 👀 Evidence
		
Prod not running from a feature branch: https://github.com/amido/stacks-dotnet/actions/runs/2455620018
Prod shown as running and deploying OK (albeit without conditions so that it only runs on master/main) : https://github.com/amido/stacks-dotnet/actions/runs/2447679894
		 
#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -183,5 +182,96 @@ jobs:
       - name: Publish Manifest File
         uses: actions/upload-artifact@v3
         with:
-          name: k8s_manifest
+          name: k8s_manifest_nonprod
+          path: ${{ env.K8S_RESOURCE_FILE }}
+
+  AppInfraProd:
+    if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/main'
+    needs: DeployDev
```
Comment (ElvenSpellmaker) [inline .github/workflows/ci.yml:190]: Dev won't deploy on main/master?

Comment (williamayerst) [reply .github/workflows/ci.yml:190]: Dev is built and deployed to on any push, but prod is only deployed to wiith main/master branch builds. This is aligned with the ADO criteria, right?

Comment (ElvenSpellmaker) [reply .github/workflows/ci.yml:190]: Currently Stacks only builds dev on branches, not master/main.

Comment (williamayerst) [reply .github/workflows/ci.yml:190]: The current condition for Dev in the air-api.yml pipeline has no criteria for build/appinfraDev/deployDev. I am happy to implement those criteria if you feel strongly, and amend the criteria in the GH pipeline. Thoughts?

**Review by dibyadhar — APPROVED** [review]

**Pull request merged**